src/vdaf.rs

```diff
@@ -68,7 +93,7 @@ pub trait Vdaf: Clone + Debug {
     type VerifyParam;
 
     /// An input share sent by a Client.
-    type InputShare: Clone + Debug + Serialize + DeserializeOwned;
+    type InputShare: Clone + Debug;
 
     /// An output share recovered from an input share by an Aggregator.
     type OutputShare: Clone + Debug + Serialize + DeserializeOwned;
```

Comment:
I think we should also remove the `Serialize + Deserialize` trait bound here, remove the derived traits from `struct OutputShare`, and provide codec routines. Same goes for `trait Aggregatable` and `struct AggregateShare`. I would expect neither `vdaf.rs` nor `vdaf/prio3.rs` to need to `use serde` anymore.

Reply:
Removed the trait bounds. I'd prefer to hold off on adding serialization for this stuff until one of us actually needs it. The reason is that, in my implementation, `OutputShare` never gets serialized (though `AggregateShare` does).

Reply:
`OutputShare` will get serialized in my implementation, because I run the collector separately from the leader and helper. However I can add that serialization myself later if you're very keen to get this merged.

Reply:
Either way!

Reply:
I'll merge this now and we can add this later. Note that, by the end of next week, my hope is that the VDAF doc will have specified all of this stuff.